src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
 
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: () => number,
+}
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const longest = [a, b, c];
+    const max: number = Math.max(...longest);
 
+    longest.splice(longest.indexOf(max), 1);
+
+    if ((a <= 0 || b <= 0 || c <= 0) || max >= (longest[0] + longest[1])) {
+      throw new Error('Length less than zero');
+    }
```

Comment:
You have the separate condition for it

```suggestion
    if ((a <= 0 || b <= 0 || c <= 0)) {
      throw new Error('Length less than zero');
    }
```